### ISSUE TITLE
Word-breaking in columns in version 3.12 has become worse

### ISSUE THREAD

**Issue by Landcross** [sqlitebrowser/sqlitebrowser#2291]
So, I just installed the new release version 3.12. It has lots of great new features, but there's one thing that immeadly caught my eye. The word-breaking in the new version has changed and, in my opinion, it's gotten worse.

In previous versions, there wasn't really any word-breaking to begin with, the text in a column just... stopped at the end of the column. The new version seems to implement some kind of word break.

Here's an example:
![image](https://user-images.githubusercontent.com/12005521/85197052-e274a300-b2de-11ea-9a9a-e43eb7efa18b.png)
![image](https://user-images.githubusercontent.com/12005521/85197055-eaccde00-b2de-11ea-86a4-1877b5747831.png)

As soon as the column is just a tiny bit too small to show the entire text, it breaks (in this case at the - character) and there's boatloads of whitespace.

I've looked in the settings to see if this is maybe something configurable, but I couldn't find anything. Please lemme know if I just overlooked it, otherwise I think this change should be reverted or made into something configurable.

Thanks!

**Comment by justinclift** [comment]
Interesting.  Wonder if that's related to the Qt wrapping changes a while back...

@mgrojo Possibly your kind of thing?

**Comment by chrisjlocke** [comment]
The previous issue was all text squished in one box (so no wrapping or elipses) while this seems a change in the word-break behaviour (which characters it breaks on)
Its whether this can be tweaked at all...

**Comment by Landcross** [comment]
Changing the row height might help, but in my opinion this is a work-around, not a fix/improvement.

Just for good measure, I made a comparison between 3.11 and 3.12. This is 3.11:
![3 11](https://user-images.githubusercontent.com/12005521/85200467-02649080-b2f8-11ea-9739-69905f5f9441.gif)

And this is 3.12:
![3 12](https://user-images.githubusercontent.com/12005521/85200471-085a7180-b2f8-11ea-93ce-bd0ccc29f2f8.gif)

The word-break in 3.12 might make more sense for actual text, but for a lot of data that is not text/words like the one in my example, it's just annoying. Again, at least in my personal opinion.

**Comment by justinclift** [comment]
Yeah.  The behaviour there is driven by the underlying Qt library we use.  They had an issue last time around that did weird things with text wrapping.  We filed a bug about that, and they did get it fixed some time later.

It sounds like this change in behaviour is either due to us updating to a newer Qt version (5.12.8) for the Windows and macOS builds, or perhaps due to Qt changing how the word breaking is done.

We can look for potential knobs to turn in Qt.  It might turn out to be something simple (hopefully!).  There's also a slightly newer Qt release out now (5.12.9) which **may** change that (unlikely).  Alternatively, there's a Qt 5.15 release series available now too.  That would be another potential thing we could look at. :smile:

**Comment by mgrojo** [comment]
I don't reproduce it under Ubuntu (Qt 5.9), so it's probably a change in behaviour in Qt. I haven't found a direct way to change the wrapping mode. In fact, in this case of a cell height of one line, the text should not be wrapped and display as many characters as possible in the line, concatenated with the ellipsis, indicating that it is incomplete. My Qt version wraps after the first hyphen (using your data) only when there is enough height to show the remaining text in the next line. This might be considered a bug. I wonder if that was introduced by the correction of the bug that we reported.

**Comment by chrisjlocke** [comment]
> but in my opinion this is a work-around, not a fix/improvement

Apologies - I did suggest it as a workaround only, and due to the limited access we have with the Qt control that this occurs in, may be the best we can do in the short term.  It will be looked at though in more depth for a permanent fix.

**Comment by justinclift** [comment]
Yeah, we'd probably want to check if it still happens with Qt 5.15.  If it does we could file a bug and see what happens.
